### PR TITLE
Fix typos in warnings, docstrings, comments and text files

### PR DESCRIPTION
**Pull request by abstractee**
This corrects some typos in comments, docstrings and text files.

### CODE REVIEW

**Comment by Wulian233** [comment]
Do not modify the idle help.html, as it is automatically generated and the typo was fixed 4 months ago. 
https://github.com/python/cpython/commit/e7ba6e9dbe5433b4a0bcb0658da6a68197c28630


"referencable" and "referenceable" are both acceptable spellings. Let's just leave this file as it is. https://en.wiktionary.org/wiki/referencable
https://github.com/python/cpython/pull/123201#discussion_r1724880043

File: Include/refcount.h [hunk 1]
```diff
@@ -401,7 +401,7 @@ static inline Py_ALWAYS_INLINE void Py_DECREF(PyObject *op)
  * There are cases where it's safe to use the naive code, but they're brittle.
  * For example, if `op` points to a Python integer, you know that destroying
  * one of those can't cause problems -- but in part that relies on that
- * Python integers aren't currently weakly referencable.  Best practice is
+ * Python integers aren't currently weakly referenceable.  Best practice is
```
Comment (Wulian233) [inline Include/refcount.h]:
```suggestion
 * Python integers aren't currently weakly referencable.  Best practice is
```

File: Lib/idlelib/help.html [hunk 2]
```diff
@@ -694,7 +694,7 @@ <h3>Shell window<a class="headerlink" href="#shell-window" title="Permalink to t
 <p>The editing features described in previous subsections work when entering
 code interactively.  IDLE’s Shell window also responds to the following:</p>
 <ul class="simple">
-<li><p><kbd class="kbd compound docutils literal notranslate"><kbd class="kbd docutils literal notranslate">C</kbd>-<kbd class="kbd docutils literal notranslate">c</kbd></kbd> attemps to interrupt statement execution (but may fail).</p></li>
+<li><p><kbd class="kbd compound docutils literal notranslate"><kbd class="kbd docutils literal notranslate">C</kbd>-<kbd class="kbd docutils literal notranslate">c</kbd></kbd> attempts to interrupt statement execution (but may fail).</p></li>
```
Comment (Wulian233) [inline Lib/idlelib/help.html]:
```suggestion
<li><p><kbd class="kbd compound docutils literal notranslate"><kbd class="kbd docutils literal notranslate">C</kbd>-<kbd class="kbd docutils literal notranslate">c</kbd></kbd> attemps to interrupt statement execution (but may fail).</p></li>
```

**Review by Wulian233 — CHANGES_REQUESTED** [review]

File: Lib/_pyio.py [hunk 3]
```diff
@@ -315,7 +315,7 @@ class IOBase(metaclass=abc.ABCMeta):
 
     This class provides dummy implementations for many methods that
     derived classes can override selectively; the default implementations
-    represent a file that cannot be read, written or seeked.
+    represent a file that cannot be read, written or sought.
```
Comment (picnixz) [inline Lib/_pyio.py]: While I understand this change, I think we can use `seeked` here. One existing wording is https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/seeked_event which relates to `seek` operations.

File: Include/refcount.h (hunk 1 above)
Comment (abstractee) [reply Include/refcount.h]: Thanks for the feedback. Corrected.

File: Lib/idlelib/help.html (hunk 2 above)
Comment (abstractee) [reply Lib/idlelib/help.html]: Removed the file from the commit. Thank you.

File: Include/refcount.h (hunk 1 above)
Comment (nineteendo) [reply Include/refcount.h]: ~~Aren't both of these changes correct?~~

**Review by jaraco — APPROVED** [review]
LGTM (importlib/resources/_common.py).

**Comment by vstinner** [comment]
Merged, thanks. I don't think that it's worth it to backport the change.

**Comment by corneliusroemer** [comment]
I'm getting the "ambiguious" typo in Python 3.13 - might be worth backporting as a bugfix.

Just checking for "year specified is ambiguious and" on Github shows lots of exact matches: https://github.com/search?q=%22year+specified+is+ambiguious%22&type=code

**Pull request merged**